src/pextlib1.0/Pextlib.c

```diff
@@ -663,38 +804,52 @@ typedef struct volcaps {
  * Default function for determining the FS case sensitivity on Darwin.
  * Using getattrlist().
  */
-int fs_case_sensitive_darwin(const char *path) {
-    /* TODO: cache mount point CS value. */
+int fs_case_sensitive_darwin(Tcl_Interp *interp, const char *path, mount_cs_cache_t *cache) {
     int ret = -1;
 
-    struct statfs f = { 0 };
-    struct attrlist attrlist = { 0 };
-    volcaps_t volcaps = { 0 };
-
     if (!path) {
         return ret;
     }
 
-    if (-1 == statfs(path, &f)) {
+    char *mntpoint = get_mntpoint(path);
+
+    if (!mntpoint) {
         return ret;
     }
 
+    struct attrlist attrlist = { 0 };
+    volcaps_t volcaps = { 0 };
+
     attrlist.bitmapcount = ATTR_BIT_MAP_COUNT;
     attrlist.volattr = ATTR_VOL_CAPABILITIES;
 
-    if (-1 == getattrlist(f.f_mntonname, &attrlist, &volcaps, sizeof(volcaps), 0)) {
+    if (-1 == getattrlist(mntpoint, &attrlist, &volcaps, sizeof(volcaps), 0)) {
```

Comment:
Shouldn't you first check `lookup_mount_cs_cache()` before calling the potentially more expensive `getattrlist(2)`?